src/browser-frontend/views/browser/chrome/page.jsx

```diff
@@ -38,6 +44,10 @@ export default class Page extends PureComponent {
     this.props.dispatch(WebContentsActions.events.pageDidMount({ pageId: this.props.pageId }));
   }
 
+  componentDidUpdate() {
+    this.props.dispatch(WebContentsActions.events.pageDidChangeImpl({ pageId: this.props.pageId }));
```

Comment:
Currently, all changes result in a page implementation change if I'm not mistaken. But I agree, in the future, as we possibly add more props to this component, then this event would be misleading. Will change.